model/task/task.go

```diff
@@ -2111,6 +2111,11 @@ func (t *Task) MarkEnd(finishTime time.Time, detail *apimodels.TaskEndDetail) er
 		})
 }
 
+// HasOrWillRun returns whether the task has run or will run in the future.
+func (t *Task) HasOrWillRun() bool {
+	return t.Activated || evergreen.IsValidTaskEndStatus(t.Status) || t.Status == evergreen.TaskStarted
```

Comment:
How do we want to treat tasks that are activated but Blocked?

Reply:
I think it should still return true. I'll move this in the other function but essentially if it is activated but blocked then the task group should not restart just yet (until that task that is activated but blocked is ran)

Reply:
`if it is activated but blocked then the task group should not restart just yet`
in that case, don't we want `t.Activated` to be `(t.Activated && !t.Blocked())`?

Reply:
discussed offline

Reply:
discussed switching to only checking t.Activated.